src/components/dashboard/main-content/course-enrollments/data/hooks.js

```diff
@@ -191,14 +187,33 @@ export const useCourseUpgradeData = ({
       }
     };
 
+    // Don't do anything if the course is not part of the enterprise's catalog
+    if (!containsContentItems) {
+      setIsLoading(false);
+      return;
+    }
+
     fetchData();
-  }, [courseRunKey, enterpriseId, subscriptionLicense, location, couponCodes]);
+
+    // Currently can only check against canRedeemMetadata for one key at a time
+    if (canRedeemMetadata.length !== 0) {
+      if (canRedeemMetadata[0].canRedeem) {
+        setSubsidyForCourse(canRedeemMetadata[0].redeemableSubsidyAccessPolicy);
```

Comment:
I believe this may need to be transformed similar to the other data set via `setSubsidyForCourse` (e.g., see `getSubsidyToApplyForCourse` specifically).